salt/states/azurearm_dns.py

```diff
@@ -590,32 +577,32 @@ def record_set_present(
                     if not isinstance(record, list):
                         ret[
                             "comment"
-                        ] = "{0} record information must be specified as a list of dictionaries!".format(
+                        ] = "{} record information must be specified as a list of dictionaries!".format(
                             record_str
                         )
                         return ret
                     local, remote = [
                         sorted(config) for config in (record, rec_set[record_str])
                     ]
-                    for idx in six_range(0, len(local)):
-                        for key in local[idx]:
-                            local_val = local[idx][key]
-                            remote_val = remote[idx].get(key)
-                            if isinstance(local_val, six.string_types):
+                    for val in local:
+                        for key in val:
+                            local_val = val[key]
+                            remote_val = val.get(key)
```

Comment:
**suggestion** there's something off here. The original was `remote[idx].get(key)`

Perhaps that should be `for local_val, remote_val in zip(local, remote)` or izip_longest?